File: app/src/main/java/com/example/jaguar/weather/model/WeatherModel.java
package com.example.jaguar.weather.model;

import com.example.jaguar.weather.model.network.RetrofitConnection;
import java.util.List;

public class WeatherModel implements WeatherContract.WeatherModel {
    RetrofitConnection ret = new RetrofitConnection();
    @Override
    public void updateWeather(String str, final CallBack call) {
        ret.getWeatherFullTime(str, new CallBack() {
            @Override
            public void UpdateWeather(List<WeatherObj> objs) {
                call.UpdateWeather(objs);
            }
        });
    }
}

File: app/src/main/java/com/example/jaguar/weather/model/WeatherModel.kt
package com.example.jaguar.weather.model

import com.example.jaguar.weather.interfaces.CallBack
import com.example.jaguar.weather.common.WeatherObj
import com.example.jaguar.weather.common.ParseWeather

class WeatherModel{
    var ret = ParseWeather()
    fun updateWeather(str: String, call: CallBack) {
        ret.getWeatherFullTime(str, object : CallBack {
            override fun UpdateWeather(objs: MutableList<WeatherObj>) {
               call.UpdateWeather(objs)
            }
        })
    }
}